bgpd/bgp_btoa.c

```diff
@@ -217,6 +217,11 @@ int main(int argc, char **argv)
 
 		printf("len: %zd\n", len);
 
+		if (len > ((size_t)-1) - 12 || stream_get_size(s) < len + 12) {
+			printf("ENDOF FILE 1");
+			break;
+		}
+
 		fread(s->data + 12, len, 1, fp);
```

Comment:
Shouldn't we be using the stream read function(s) for reading from a file descriptor (`stream_read_try`, `stream_read`) that do bounds checking internally rather than `fread` directly into stream memory?

Reply:
Well, I guess that the purpose is to simplify buffering (calling stream_reset() and accessing directly to the beginning of the buffer with s->data). I.e. unless I missed something, the "stream" functions don't seem to be that smart or the usage is limited/wrong across the main() function of bgpd/bgp_btoa.c file.

Reply:
(e.g. if you use a big buffer, logic for adding message fragmentation should be added, which could be probably more complicated because the stream functions not being able to do the I/O -unless I missed something-)